cscs-checks/libraries/math/dgemm.py

```diff
@@ -1,31 +1,30 @@
-import os
+import reframe as rfm
 import reframe.utility.sanity as sn
 
-from reframe.core.pipeline import RegressionTest
 
-
-class DGEMMTest(RegressionTest):
-    def __init__(self, name, **kwargs):
-        super().__init__('DGEMM_' + name, os.path.dirname(__file__), **kwargs)
+class DGEMMTest(rfm.RegressionTest):
+    def __init__(self, name):
+        super().__init__()
+        self.name = 'DGEMM_' + name
         self.descr = 'DGEMM performance test'
         self.sourcepath = 'dgemm.c'
         self.executable_opts = ['5000', '5000', '5000']
-        self.cflags = None
-        self.ldflags = None
+        self.build_system = 'SingleSource'
+        self.build_system.cflags = self.cflags
+        self.build_system.ldflags = self.ldflags
```

Comment:
I don't like this design. The subclasses should set directly the flags the need.